frontend/src/app/main/ui/dashboard/grid.cljs

```diff
@@ -312,24 +312,25 @@
 
         on-menu-click
         (mf/use-fn
-         (mf/deps file selected?)
+         (mf/deps file selected? is-library-view)
          (fn [event]
            (dom/stop-propagation event)
            (dom/prevent-default event)
-           (when-not selected?
-             (when-not (kbd/shift? event)
-               (st/emit! (dd/clear-selected-files)))
-             (st/emit! (dd/toggle-file-select file)))
-
-           (let [client-position (dom/get-client-position event)
-                 position (if (and (nil? (:y client-position)) (nil? (:x client-position)))
-                            (let [target-element (dom/get-target event)
-                                  points         (dom/get-bounding-rect target-element)
-                                  y              (:top points)
-                                  x              (:left points)]
-                              (gpt/point x y))
-                            client-position)]
-             (st/emit! (dd/show-file-menu-with-position file-id position)))))
+           (when-not is-library-view
```

Comment:
This is working but is getting a bit hard to read, maybe we can refactor it a little bit.